### ISSUE TITLE
Validate LR scheduler config

### ISSUE THREAD

**Issue by jpdunc23** [ai2cm/ace#674]
`SchedulerConfig.build` is not called until runtime so we can't currently catch a misconfigured `torch.optim.lr_scheduler._LRScheduler` in `validate_config.py`.

See https://github.com/ai2cm/full-model/pull/2108/files#r2370200885
